velox/exec/tests/TableScanTest.cpp

```diff
@@ -1499,6 +1499,15 @@ TEST_F(TableScanTest, partitionedTableDoubleKey) {
   testPartitionedTable(filePath->path, DOUBLE(), "3.5");
 }
 
+TEST_F(TableScanTest, partitionedTableDateKey) {
+  auto rowType = ROW({"c0", "c1"}, {BIGINT(), DOUBLE()});
+  auto vectors = makeVectors(10, 1'000, rowType);
+  auto filePath = TempFilePath::create();
+  writeToFile(filePath->path, vectors);
+  createDuckDbTable(vectors);
+  testPartitionedTable(filePath->path, DATE(), "2023-10-27");
```

Comment:
> These was a deprecated query config isIso8601, but could not be leaked to connector as previous discussion #7084 (comment) pointed out

I'm remembering now. What other date formats are supported? Can we add test cases to make sure all supported formats work and different format are supported based on what version of CAST is registered?

Reply:
> Why can't we use a lower-level utility?

Hi @mbasmanova, I took a further look here. The format of partitionKeys can be controlled by Gluten, so if we ensure that compatible strings are generated as partition values, maybe we don't need to bring more complexity here. For date format `YYYY-MM-DD`, `castFromDateString` with true for `isIso8601` should be enough. Do you think it makes sense?

Reply:
I notice we can control in Gluten to generate well-formatted strings are partition values. Maybe we can keep using light-weight conversion here.

Reply:
> Maybe we can keep using light-weight conversion here.

That would be great.

Reply:
Updated. Could you help review again? Thanks.